PracticeGit/Test.c

```diff
@@ -1,13 +1,125 @@
 // ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
-// ------------------------------------------------------------------
-// test.c
 // ------------------------------------------------------------------------------------------------
+// test.c -  Program that Executes test cases and checks the user input for sort & search.
+// ------------------------------------------------------------------------------------------------
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#define MAXSIZE 10
+#define RED     "\033[1;31m"
+#define GREEN   "\033[1;32m"
+#define RESET   "\033[0m"
+
+/// <summary> Prints the elements of an array. </summary>
+void PrintArray (int array[], int size);
+
+/// <summary> Runs test cases for the QuickSort function. </summary>
+void RunQuickSortTest ();
+
+/// <summary> Gets user input for an array, sorts it, and performs a binary search. </summary>
+void GetUserInput ();
+
+typedef struct {
+   int array[MAXSIZE];
+   int size;
+} TestCase;
+
+void PrintArray (int array[], int size) {
+   for (int i = 0; i < size; i++) printf ("%5d", array[i]);
+}
+
+void RunQuickSortTest () {
+   TestCase testCases[] = {
+       {{20, 49, 11, 3, 89, 75, 11, 12, 55, 1}, 10},
+       {{2, 2, 2, 2, 2, 2, 2}, 7},
+       {{9}, 1},
+       {{3, -1, -7, 5, 2, 0, -7}, 7},
+       {{1, 2, 3, 4, 5, 6}, 6},
+       {{0, -1, -2, -3, -4, -5}, 6},
+       {{36, 69}, 2},
+       {{22, 35, 32, 39, 46, 54, 55}, 7},
+       {{12, 13, 14, 15, 16, 11}, 6}
+   };
+   int expectedResults[][MAXSIZE] = {
+       {1, 3, 11, 11, 12, 20, 49, 55, 75, 89},
+       {2, 2, 2, 2, 2, 2, 2},
+       {9},
+       {-7, -7, -1, 0, 2, 3, 5},
+       {1, 2, 3, 4, 5, 6},
+       {-5, -4, -3, -2, -1, 0},
+       {36, 69},
+       {22, 32, 35, 39, 46, 54, 55},
+       {11, 12, 13, 14, 15, 16}
+   };
+   for (int i = 0; i < 9; i++) {
+      int n = testCases[i].size;
+      int currentTest[MAXSIZE] = { 0 };
+      memcpy (currentTest, testCases[i].array, n * sizeof (int));
+      printf ("\nTestcase %d\nUnsorted array: ", i + 1);
+      PrintArray (testCases[i].array, n);
+      QuickSort (currentTest, 0, n - 1);
+      printf ("\nSorted array:   ");
+      PrintArray (currentTest, n);
+      const char* result = (memcmp (currentTest, expectedResults[i], n * sizeof (int)) == 0) ? GREEN"Pass"RESET : RED"Fail"RESET;
+      printf ("\nResult: %s\n", result);
+   }
+}
 
-#include "Program.h"
+void GetUserInput () {
+   int array[MAXSIZE];
+   int size;
```

Comment:
Initializations with the same data type can be done in a single line using a comma. Apply this change wherever applicable.